framework/contracts/account/src/contract.rs

```diff
@@ -283,12 +280,18 @@ pub fn execute(mut deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg)
                             .add_messages(msgs),
                     )
                 }
-                ExecuteMsg::ModuleAction { msgs } => {
-                    execute_module_action(deps, info, msgs).map_err(AccountError::from)
+                ExecuteMsg::AccountActions { msgs } => {
+                    execute_account_action(deps, &info.sender, msgs).map_err(AccountError::from)
                 }
-                ExecuteMsg::ModuleActionWithData { msg } => {
-                    execute_module_action_response(deps, info, msg).map_err(AccountError::from)
+                ExecuteMsg::AccountActionWithData { msg } => {
+                    execute_account_action_response(deps, &info.sender, msg)
+                        .map_err(AccountError::from)
                 }
+
+                ExecuteMsg::ExecOnModule {
+                    module_id,
+                    exec_msg,
+                } => exec_on_module(deps, info, module_id, exec_msg).map_err(AccountError::from),
```

Comment:
It looks like one module can call base message on another module. It's fine, as long as user trust base method of the adapter, right?

In case it's not clear how:
- Admin calls `admin_on_module` on whitelisted module with custom base handler
- Now this module can call base via `exec_on_module` on any module or modules (`ADMIN_CALL_TO_CONTEXT` will not be cleared until this module execution is done)

P.S. Sorry if I ruined your backdoor :rofl:

Reply:
Oh, nice! Missed that there's mermaid graphs as well, thank you.